D/DynarePreprocessor/build_tarballs.jl

```diff
@@ -0,0 +1,46 @@
+using BinaryBuilder
+
+
+name = "DynarePreprocessor"
+version = v"4.8.0"
+sources = [
+    GitSource("https://git.dynare.org/Dynare/preprocessor.git", "a8fce06dc46ca09329e6899e1fde47f2cd81809b"),
+]
+
+script = raw"""
+HostBuildDependency("boost_jll")
+HostBuildDependency("Bison_jll")
+HostBuildDependency("flex_jll")
+
+cd ${WORKSPACE}/srcdir/preprocessor
+
+# remove -lstdc++fs in Makefile.am
+sed s/-lstdc++fs// -i src/Makefile.am
+
+autoreconf -si
+
+# use non default gcc for apple and freebsdn
+if [[ "${target}" == *-freebsd* ]] || [[ "${target}" == *-apple-* ]]; then
+    CC=gcc
```

Comment:
I think you must export these variables for them to have any effect on child processes, i.e.:
```suggestion
    export CC=gcc
```
And the same for CXX.

Alas, I think it'd be better to stick to the default compilers and instead patch the source code mistake I pointed out earlier. There are plenty builders using patch files that you can look at to get an idea how to do it. Just search for `atomic_patch`